src/surface/polygon_soup_mesh.cpp

```diff
@@ -120,6 +148,177 @@ void PolygonSoupMesh::readMeshFromFile(std::string filename) {
   }
 }
 
+// Assumes that first line has already been consumed
+void PolygonSoupMesh::readMeshFromAsciiStlFile(std::ifstream& in) {
+  std::string line;
+  std::stringstream ss;
+  size_t lineNum = 1;
+
+  auto assertToken = [&](const std::string& expected) {
+    std::string token;
+    ss >> token;
+    if (token != expected) {
+      std::cerr << "Error on line " << lineNum << ". Expected \"" << expected << "\" but token \"" << token << "\""
+                << std::endl;
+      std::cerr << "Full line: \"" << line << "\"" << std::endl;
+      exit(1);
```

Comment:
Should probably `throw std::runtime_error()` here, rather than `exit(1)`. A user might want to catch an exception if loading fails.

Also, `std::cerr` is fine, but I tend to prefer using `std::cout` everywhere. The reason why is that when we intermingle `std::cout` and `std::cerr`, the print statements can end out of order in the terminal, which gets confusing. Using only `std::cout` keeps a global ordering.

Reply:
I updated this to print everything in the `std::runtime_error`.